src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -287,30 +243,25 @@ public bool TryCreateSolutionContext(string solutionDirectory, out IVsPathContex
                 trie);
         }
 
-        private async Task<IVsPathContext> GetPathContextForPackagesConfigAsync(
-            IVsProjectAdapter vsProjectAdapter, CancellationToken token)
+        private async Task<IVsPathContext> GetPathContextFromPackagesConfigAsync(
+            NuGetProject nuGetProject, CancellationToken token)
         {
-            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-            var props = new Dictionary<string, object>();
-            props.Add(NuGetProjectMetadataKeys.Name, Path.GetFileNameWithoutExtension(vsProjectAdapter.FullProjectPath));
-            props.Add(NuGetProjectMetadataKeys.TargetFramework, await vsProjectAdapter.GetTargetFrameworkAsync());
-
-            var packagesProject = new PackagesConfigNuGetProject(vsProjectAdapter.ProjectDirectory, props);
+            var msbuildNuGetProject = nuGetProject as MSBuildNuGetProject;
+            if (msbuildNuGetProject == null)
+            {
+                return null;
+            }
 
-            var packagesFolderPath = PackagesFolderPathUtility.GetPackagesFolderPath(_solutionManager.Value, _settings.Value);
-            var folderProject = new FolderNuGetProject(packagesFolderPath);
+            var packageReferences = await msbuildNuGetProject.GetInstalledPackagesAsync(token);
```

Comment:
Since this is the same method as before, just on a different model, was it just the props reading that was causing the delay?

Reply:
No it was `vsProjectAdapter.GetTargetFrameworkAsync()` which was causing delay since it required to load `BuildPropertyStorage` instance...

Reply:
Ah, makes sense then.